src/v2/auth.rs

```diff
@@ -266,9 +267,13 @@ impl Client {
             authentication_header,
         )? {
             WwwAuthenticateHeaderContent::Basic(_) => {
-                let basic_auth = BasicAuth {
-                    user: credentials.0,
-                    password: Some(credentials.1),
+                let basic_auth = {
+                    let (user, password) =
+                        credentials.expect("cannot authenticate without credentials");
+                    BasicAuth {
+                        user,
+                        password: Some(password),
+                    }
                 };
```

Comment:
```suggestion
                let basic_auth = credentials
                    .map(|(user, password)| BasicAuth {
                        user,
                        password: Some(password),
                    })
                    .ok_or("cannot authenticate without credentials")?;
```

Reply:
Thanks, I'm still getting the ropes of Rust coming in from the C++/Scala world